src/write_gcm.c

```diff
@@ -2275,59 +2274,155 @@ static monitored_resource_t *monitored_resource_create_from_fields(
   // Changes va_list into label_t array
   va_start(ap, project_id);
   for (int i = 0; i < num_labels; i++) {
-    labels[i].key = va_arg(ap, const char*);
-    labels[i].value = va_arg(ap, const char*);
+    labels[i].key = va_arg(ap, char*);
+    labels[i].value = va_arg(ap, char*);
   }
   va_end(ap);
   return monitored_resource_create_from_array(type, project_id, labels,
                                               num_labels);
 }
 
+typedef struct label_node_s
+{
+  label_t label;
+  struct label_node_s *next;
+} wg_mr_label_node_t;
+
+typedef struct {
+  int processing_labels;
+  int total_labels;
+  char *last_key;
+  char *type;
+  wg_mr_label_node_t *head;
```

Comment:
How about `llist_t *label_list`? Then you also won't need `total_labels` (or the `label_node_s` struct).